tools: update marked dependency

firedfox opened the pull request:
##### Checklist

- [x] tests and code linting passes
- [x] the commit message follows commit guidelines

##### Affected core subsystem(s)

tools

##### Description of change

Update module `tools/doc/node_modules/marked`. Customize renderer to remove id from heading.

Diff result for all.html <http://106.187.89.52/tmp/diff-683f2a7-doc-html-all.html>
Diff result for all.json <http://106.187.89.52/tmp/diff-683f2a7-doc-json-all.html>

The newly updated marked module fixed several problems in the doc.

###### Removed unnecessary `</p><p>`

In `addons.md`:
```md
Please see the examples below for further information or
<https://github.com/arturadib/node-qt> for an example in production.
```

HTML result from:
```html
<p>Please see the examples below for further information or
</p>
<p><a href="https://github.com/arturadib/node-qt">https://github.com/arturadib/node-qt</a> for an example in production.

</p>
```

to:
```html
<p>Please see the examples below for further information or
<a href="https://github.com/arturadib/node-qt">https://github.com/arturadib/node-qt</a> for an example in production.</p>
```

###### Fixed unordered list

In `util.md`:
```md
`util.inspect.styles` is a map assigning each style a color
from `util.inspect.colors`.
Highlighted styles and their default values are:
 * `number` (yellow)
 * `boolean` (yellow)
 * `string` (green)
 * `date` (magenta)
 * `regexp` (red)
 * `null` (bold)
 * `undefined` (grey)
 * `special` - only function at this time (cyan)
 * `name` (intentionally no styling)
```

HTML result from:
```html
<p><code>util.inspect.styles</code> is a map assigning each style a color
from <code>util.inspect.colors</code>.
Highlighted styles and their default values are:
 <em> <code>number</code> (yellow)
 </em> <code>boolean</code> (yellow)
 <em> <code>string</code> (green)
 </em> <code>date</code> (magenta)
 <em> <code>regexp</code> (red)
 </em> <code>null</code> (bold)
 <em> <code>undefined</code> (grey)
 </em> <code>special</code> - only function at this time (cyan)
 * <code>name</code> (intentionally no styling)

</p>
```

to:
```html
<p><code>util.inspect.styles</code> is a map assigning each style a color
from <code>util.inspect.colors</code>.
Highlighted styles and their default values are:</p>
<ul>
<li><code>number</code> (yellow)</li>
<li><code>boolean</code> (yellow)</li>
<li><code>string</code> (green)</li>
<li><code>date</code> (magenta)</li>
<li><code>regexp</code> (red)</li>
<li><code>null</code> (bold)</li>
<li><code>undefined</code> (grey)</li>
<li><code>special</code> - only function at this time (cyan)</li>
<li><code>name</code> (intentionally no styling)</li>
</ul>
```

###### Removed redundant new lines

In `addons.md`:
```md
To test:
```

HTML result from:
```html
<p>To test:

</p>
```

to:
```html
<p>To test:</p>
```

###### code element class name changes

It changed the class name for `code` element from `cpp / js / javascript` to `lang-cpp / lang-js / lang-javascript`. It does not matter because there is no style defined for these class names.

silverwind commented:
@nodejs/documentation

addaleax commented on tools/doc/html.js:
```diff
@@ -8,6 +8,15 @@ var typeParser = require('./type-parser.js');
 module.exports = toHTML;
 
+// customized heading without id attribute
+var renderer = new marked.Renderer();
```
`const`?

silverwind replied:
Wouldn't really fit the style of this file. I guess we shoul to es6ify the doctool later.

thefourtheye replied:
Two lines below this, template strings are used :P

firedfox replied:
Yes it's a problem. I think I'd better change the template string to string replace in order to keep the style consistent.

addaleax replied:
fwiw, I also used template strings in #6495, I don’t really think that’s a problem. And sorry for the merge conflict, btw, but that should be limited to the package.json.

firedfox replied:
It doesn't matter. I rebased upstream and resolved the conflict. 
I just changed the template to string add before I saw your latest comment :D
I think either is OK. At least the style is consistent now. We can es6ify it later.

addaleax commented on tools/doc/json.js:
```diff
@@ -7,6 +7,15 @@ module.exports = doJSON;
 
 var marked = require('marked');
 
+// customized heading without id attribute
```
Does this have any effect on the JSON output?

firedfox replied:
It's all the same as the HTML output. I put diff results of all.html and all.json in the description of change.

silverwind commented:
Let's see if the doctool tests pass:

https://ci.nodejs.org/job/node-test-pull-request/2500/

firedfox commented:
There seems to be something wrong with the armv8-ubuntu1404 environment.
![image](https://cloud.githubusercontent.com/assets/171705/15061937/918bbd62-136d-11e6-9ae1-a6b4c53fee7a.png)

silverwind commented:
Probably nothing to worry about, I bet it'll be green next run. Here's another CI:https://ci.nodejs.org/job/node-test-pull-request/2523/

LGTM

addaleax commented:
CI is red, example failure: https://ci.nodejs.org/job/node-test-commit-arm/3159/nodes=armv7-wheezy/tapTestReport/test.tap-1068/

That’s just whitespace changes in the test results, though. It may be nice to do something there like we do for the HTML output tests, i.e. strip whitespaces in both expected and actual output and compare that.

firedfox commented:
@addaleax It's my bad. I modified the test case to make my local tests pass but forgot to commit it. It's updated now. I just removed extra `\n`s from the expected data. Those `\n`s are bugs of the old version marked module. I think they will not change in future. So we don't have to take more effort to strip them from json objects.

addaleax commented:
LGTM pending CI: https://ci.nodejs.org/job/node-test-commit/3215/

addaleax commented:
CI is green.

@firedfox the `Author:` field in your git commit is set to `firedfox`, but you are listed with your full name in the `AUTHORS` file. If you want, this can be fixed in when landing the commit?

firedfox commented:
Oh yes, please change the `Author` field to my full name. Thank you!

addaleax commented:
Landed in 3f69ea53fd800e1f884834e5d032dd89e58fa35c. Thanks!

You can set this via `git config [--global] user.name '…'`, btw. :)

bnoordhuis commented:
FYI, this PR looks to be breaking `tools/doc/addon-verify.js`; as a result, the embedded add-ons in the documentation are no longer built and tested.

I'm going to look into what's causing the breakage but if I can't figure it out or it's too much effort to fix, I'll be moving to revert it.

addaleax commented:
@bnoordhuis addaleax/node@6e4342fb7e8020b32f3b9696529feebb641a6179 seems to do the trick?

bnoordhuis commented:
@addaleax Yes, seems to be working.

bnoordhuis commented:
https://github.com/nodejs/node/pull/6652

bnoordhuis commented:
Release people: if this gets backported, then please also cherry-pick commit 4f925dd from https://github.com/nodejs/node/pull/6652.
